firebase-firestore/src/main/java/com/google/firebase/firestore/remote/FirestoreChannel.java

```diff
@@ -281,10 +278,24 @@ public void invalidateToken() {
     credentialsProvider.invalidateToken();
   }
 
+  public static void setClientLanguage(String languageToken) {
+    clientLanguage = languageToken;
+  }
+
+  private static String getDefaultClientLanguage() {
+    // Note: there is no good way to get the Java language version on Android
+    // (System.getProperty("java.version") returns "0", for example).
+    return "gl-java/";
+  }
+
+  private String getGoogApiClientValue() {
+    return String.format("%s fire/%s grpc/", clientLanguage, BuildConfig.VERSION_NAME);
```

Comment:
Unfortunately, the method to retrieve the gRPC version is not public in the Java SDK.